source/codegen/service_helpers.py

```diff
@@ -1,75 +1,110 @@
 import common_helpers
 
+
 def get_include_guard_name(config, suffix):
     include_guard_name = config['namespace_component'] + "_grpc" + suffix
     return include_guard_name.upper()
 
+
+def get_c_element_type_for_array_that_needs_coercion(parameter):
+    if not parameter.get('coerced', False):
+        return None
+    stripped_type = parameter['type']
+    if stripped_type.startswith('const '):
+        stripped_type = stripped_type[len('const '):]
+    if stripped_type.endswith('*'):
+        stripped_type = stripped_type[:-1]
+    elif stripped_type.endswith('[]'):
+        stripped_type = stripped_type[:-2]
+    else:
+        return None
+    return stripped_type
+
+
+def is_input_array_that_needs_coercion(parameter):
+    return common_helpers.is_input_parameter(parameter) and get_c_element_type_for_array_that_needs_coercion(parameter) is not None
+
+
+def is_output_array_that_needs_coercion(parameter):
+    return common_helpers.is_output_parameter(parameter) and get_c_element_type_for_array_that_needs_coercion(parameter) is not None
+
+
 def create_args(parameters):
     result = ''
-    is_twist_mechanism = common_helpers.has_ivi_dance_with_a_twist_param(parameters)
+    is_twist_mechanism = common_helpers.has_ivi_dance_with_a_twist_param(
```

Comment:
Is this getting auto-formatted by clang format?

I'd leave this the way you have it for this change. Can you try out "Format On Save Mode: modifications" in the future?

Reply:
Yeah, this is getting auto-formatted. I do have my Format On Save Mode set to modifications; I think in this case the indentation got messed up in a merge somehow, and I was fixing that, so maybe that's what did it?